File: frontend/src/types/invoice.ts
export interface InvoiceApi {
  id: number;
  company_id: number;
  start_date: string;
  end_date: string;
  due_date: string;
  invoice_number?: string | null;
  format_key: string;
  prepared_by_mode?: "without_signature" | "with_signature";
  signatory_id?: number | null;
  signatory_name?: string | null;
  signatory_image_path?: string | null;
  signatory_image_mime?: string | null;
  status: string;
  total_amount?: string;
  created_at?: string;
  company_name?: string;
}

export interface InvoiceCreateInput {
  companyId: number;
  startDate?: string;
  endDate?: string;
  dueDate?: string;
  invoiceNumber?: string;
  preparedByMode: "without_signature" | "with_signature";
  signatoryId?: number;
  formatKey: string;
  tripIds?: number[];
}

export interface SignatoryApi {
  id: number;
  name: string;
  signature_image_mime?: string | null;
  has_signature: boolean;
}

File: src/schemas/invoice.py
from __future__ import annotations

from datetime import date, datetime
from decimal import Decimal

from pydantic import Field

from src.schemas.common import ORMModel


class InvoiceCreate(ORMModel):
    company_id: int
    start_date: date | None = None
    end_date: date | None = None
    due_date: date | None = None
    invoice_number: str | None = Field(default=None, max_length=60)
    prepared_by_mode: str = Field(default="without_signature", max_length=25)
    signatory_id: int | None = None
    format_key: str = Field(default="standard", max_length=50)
    trip_ids: list[int] = Field(default_factory=list)


class InvoiceRead(ORMModel):
    id: int
    company_id: int
    start_date: date
    end_date: date
    due_date: date
    invoice_number: str | None
    format_key: str
    prepared_by_mode: str
    signatory_id: int | None
    signatory_name: str | None
    signatory_image_path: str | None
    signatory_image_mime: str | None
    total_amount: Decimal
    generated_at: datetime
    paid_at: datetime | None
    status: str


class InvoiceMarkPaid(ORMModel):
    paid_at: datetime | None = None


class SignatoryRead(ORMModel):
    id: int
    name: str
    signature_image_mime: str | None
    has_signature: bool
